test/extended/builds/forcepull.go

```diff
@@ -1,183 +1,179 @@
 package builds
 
-/*
-This particular builds test suite is not part of the "default" group,  because its testing
-centers around manipulation of images tags to confirm whether the `docker pull` invocation occurs
-correctly based on `forcePull` setting in the BuildConfig, and rather than spend time creating / pulling down separate, test only,
-images for each test scenario, we reuse existing images and ensure that the tests do not run in parallel, and step on each
-others toes by tagging the same, existing images in ways which conflict.
-*/
-
 import (
 	"fmt"
 
 	g "github.com/onsi/ginkgo"
 	o "github.com/onsi/gomega"
 
-	"time"
-
+	dockerClient "github.com/fsouza/go-dockerclient"
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
-var (
-	resetData map[string]string
+const (
+	buildPrefixFS = "ruby-sample-build-fs"
+	buildNameFS   = buildPrefixFS + "-1"
+	buildPrefixTS = "ruby-sample-build-ts"
+	buildNameTS   = buildPrefixTS + "-1"
+	buildPrefixFD = "ruby-sample-build-fd"
+	buildNameFD   = buildPrefixFD + "-1"
+	buildPrefixTD = "ruby-sample-build-td"
+	buildNameTD   = buildPrefixTD + "-1"
+	buildPrefixFC = "ruby-sample-build-fc"
+	buildNameFC   = buildPrefixFC + "-1"
+	buildPrefixTC = "ruby-sample-build-tc"
+	buildNameTC   = buildPrefixTC + "-1"
+
+	corruptor = "docker.io/openshift/origin-custom-docker-builder"
+
+	varSubSrc = "SERVICE_REGISTRY_IP"
+
+	s2iDockBldr       = "dock-s2i-forcepull-ruby-20-centos7"
+	s2iDockBldrPrefix = "forcepull-dock-s2i-bldr"
+
+	custBldr       = "cust-forcepull-origin-custom-docker-builder"
+	custBldrPrefix = "forcepull-cust-bldr"
 )
 
-const (
-	buildPrefix = "ruby-sample-build"
-	buildName   = buildPrefix + "-1"
-	s2iDockBldr = "docker.io/openshift/ruby-20-centos7"
-	custBldr    = "docker.io/openshift/origin-custom-docker-builder"
+var (
+	resetData  map[string]string
+	authCfg    *dockerClient.AuthConfiguration
+	fullNameDS string
+	fullNameC  string
+	tags       []string
+	varSubDest string
 )
 
+func doTest(imgName, bldPrefix, bldName, debugStr string, same bool, oc *exutil.CLI) {
+	// corrupt the builder image
+	exutil.CorruptImage(imgName, corruptor)
```

Comment:
add a verify same here to ensure the corruption was successful?  or do it in the corrupt method.